### ISSUE TITLE
3d_armor mt_0.5.0-dev support

### ISSUE THREAD

**Issue by tacotexmex** [minetest-mods/skinsdb#12]
With 3d_armor mt_0.5.0-dev branch the player stands halfway through the ground.

**Comment by bell07** [comment]
I did not tested mt_0.5.0-dev, heard about some changes for player but do not know the details.

Do you use 0.5.0-dev Minetest? Does 3d_armor mt_0.5.0-dev work for you without skinsdb?

I know the issue I need to port some functions to "player_api" if 0.5. is there.

Agree, the skinsdb is not ready for 0.5.0 yet, as the most of my mods. I use the stable engine only so I can fix it if 0.5.0 is released. But Patching ideas and Pull-Requests welcome

One try: Please replace the "skinsdb/models/**skinsdb_3d_armor_character.b3d**" file by the file [3d_armor_character.b3d](https://github.com/stujones11/minetest-3d_armor/blob/MT_0.5.0-dev/3d_armor/models/3d_armor_character.b3d) (from 3d_Armor mt_0.5.0-dev). Just copy from 3d_armor and rename / replace the file. I use the models from @stujones11 since 1.8-skins support.

**Comment by SmallJoker** [comment]
After these months of development I actually don't remember where the player collision box and its model got a new offset -- however, previously the model had its origin at 1m above ground, which is now at exactly 0m (feet position). Just a hit & miss idea: check whether player_api exists. If it exists, use the [new model ](https://github.com/stujones11/minetest-3d_armor/commit/b1bd95e6aba), if not. keep the current existing one to provide 0.4.?? compatibility.

**Comment by bell07** [comment]
@SmallJoker , is the [player_api/models](https://github.com/minetest/minetest_game/tree/master/mods/player_api/models)/character.b3d useable for 3d_armor and 1.8er Skins? Or I still need to use the 3d_armor model?

**Comment by bell07** [comment]
Just for reference, for player_api I proposed some changes for better skins support: https://github.com/minetest/minetest_game/issues/1965#issuecomment-382336806

**Comment by SmallJoker** [comment]
@bell07 Because it's skinsdb which seems to specify the used player model, the 3d_armor model with 1.8 skin support should be used to ensure the armor compatibility.

**Comment by bell07** [comment]
Initial work done in https://github.com/bell07/minetest-skinsdb/tree/mt_0_5_dev
 @tacotexmex , can you please test? I do not have 0.5 for testing (I am stable user)

**Comment by tacotexmex** [comment]
Will test, thanks alot for this effort!

**Comment by bell07** [comment]
Do you use minetest_game-0.5-dev or have the player_api mod? The "if" check for player_api to decide which implementation should be used

**Comment by tacotexmex** [comment]
I use the latest engine with the latest minetest_game. It has the player_api mod.

**Comment by bell07** [comment]
Ok, now compiled 0.5-dev to test this issue ...

Played arround and found out, if you use [playeranim](https://github.com/minetest-mods/playeranim) mod, you need to set 
```
player_model_version = default_character_v3
```
to you minetest.conf to get it working. No changes necessary in skinsdb this way.

Without playeranim my last changes does work with 3d_armor and without. Ok, with 3d_armor I nab wrong model file, updated to the right one now.

Pls. noe the current mt_0_5_dev is a temporary solution till https://github.com/minetest/minetest_game/pull/2122 is completed, so I do not merge them to minetest-mods

**Comment by tacotexmex** [comment]
I can now also get it working properly with 3d_armor(mt_0.5.0-dev)+skinsdb. However, I also use [multiskin](https://github.com/stujones11/multiskin/) mod so with this mod activated as well the bug persists. But that's an issue with multiskin, right?

**Comment by bell07** [comment]
I did not tested the skinsdb and multiskin together because I just did not understand the multiskin. So asume the mods are not compatible and maybe redundant. Most usefull things I seen in multiskin I ported to skinsdb.

What is your use-case? Why you need the multiskin?

**Comment by tacotexmex** [comment]
I added it because I thought that I needed it for the multilayered skin support. (1.8)

**Comment by bell07** [comment]
Yea, in April I ported the 1.8. skins support from multiskin to skinsdb in a way the "master" 3d_armor can be used. So just place the 1.8er Skins to the skinsdb textures folder. I recommend in addition to create preview files and meta-files to be shown in inventory.

**Comment by tacotexmex** [comment]
I see! That’s great, I’ll just remove multiskin then. 

Will mods depending on multiskin work by changing the dependency to skinsdb too perhaps? 😬

**Comment by tacotexmex** [comment]
I’m thinking of stujones11/clothing.

**Comment by bell07** [comment]
> I’m thinking of stujones11/clothing.

was long time on my todo list, started working on them now ;-)

**Comment by bell07** [comment]
The clothing support is merged to master now, the bell07/skinsdb/mt_0_5_dev is rebased against them. Stu merged the https://github.com/stujones11/clothing/pull/11
@tacotexmex so just update both mods and use clotging without multiskin ;-)

For me the clothing support is still missed in smart_inventory, so it's the next task on my list for now

**Comment by tacotexmex** [comment]
Thank you, that’s very helpful!

**Comment by bell07** [comment]
@tacotexmex , did you get managed to get the 1.8 skins working on Minetest-0.5-dev?
I tested my changes again today and got crapy skin :-(
![image](https://user-images.githubusercontent.com/4731334/42636481-35ed419c-85e9-11e8-8700-7d51a11322d7.png)

**Comment by bell07** [comment]
Ok, fixed now. Got right models from Stu

**Comment by poikilos** [comment]
> With 3d_armor mt_0.5.0-dev branch the player stands halfway through the ground.

Hmm...this is still happening in 5.0.0-dev compiled yesterday, without any mods beyond minetest_game. @bell07 still halfway underground on your fork as well. Are we calling it the same issue, making a separate issue, or waiting on https://github.com/minetest/minetest_game/pull/2122 considering conversation above? I would love to be able to use this mod before that but understand if not worth it if PR is to be merged.For testing, I have an easy 5.0.0-dev update try this linux shell script:
```bash
wget -O install-minetestserver-git-all-backends.sh https://github.com/poikilos/EnlivenMinetest/blob/master/install-minetestserver-git-all-backends.sh
chmod +x install-minetestserver-git-all-backends.sh
./install-minetestserver-git-all-backends.sh both
```
(or if you only want client and not server, put `client` instead of `both`)

**Comment by bell07** [comment]
The issue should be fixed in my fork - mt_0_5_dev Branch. To be sure you use the right version please check if your copy contains the models/skinsdb_3d_armor_character_0_5.b3d file.

You can use the next link to download the right version: https://github.com/bell07/minetest-skinsdb/archive/mt_0_5_dev.zip

**Comment by wziard** [comment]
While the player doesn't stand halfway through the ground (an improvement for sure) in your fork, I don't see the armor when worn. Should it work?

Also, the normal player model looks like there are tiny gaps along the edges.

**Comment by bell07** [comment]
@wziard, it should work.
Now I updated the mt_0_5_dev merging new changes from master. Please try again with the new version. If it still does not work, same question as for character_creator: which other skins-related mods from which do you use?

Other way is to use mods from https://github.com/bell07/minetest-player_api_modpack modpack. The pack contains 2x skins selectors: "skinsdb5" is based on "skinsdb".  "skins" is based on "simple skins" you can try too. All other mods needs to replace the "original" mods.

**Comment by wziard** [comment]
It still doesn't work.

I use 3d_armor and character_creator. Bot the latest versions from git.

**Comment by TillCoyote** [comment]
Guys, MT5 has released and my player model is still drowned in a node((

I've no blender skills so I request it again.

**Comment by bell07** [comment]
Can you pls check if the version from [my fork mt_0_5_dev branch ](https://github.com/minetest-mods/skinsdb/compare/master...bell07:mt_0_5_dev) does work for you? If yes I prepare the push to minetest-mods

**Comment by TillCoyote** [comment]
Checked it. It works with 1.0-er skins, but 1.8-er are buggy. 
![screenshot_20190309_133141](https://user-images.githubusercontent.com/13682559/54067904-f6be9900-426f-11e9-887d-7e175a210792.png)

**Comment by bell07** [comment]
After I wrote I updated the branch. I fixed this issue in the past but the change seems to be lost. So I placed again the right model file. Please check if you have the latest commit from the branch

**Comment by TillCoyote** [comment]
Succesfully it works! Thank you!

![screenshot_20190309_190233](https://user-images.githubusercontent.com/13682559/54071934-eb826200-429d-11e9-860e-2c9b9c164992.png)

**Comment by bell07** [comment]
I checked my old messages in forum and found (again) the right models. Thanks once again to stu

@TillCoyote pls. next try

**Comment by TillCoyote** [comment]
![screenshot_20190310_122812](https://user-images.githubusercontent.com/13682559/54081527-29c86180-4330-11e9-9b4f-cb8f794b8f7a.png)

@bell07 you made it cool, thanks!

Also, it probably have to support "clothing" mod but I need it no.

**Comment by bell07** [comment]
done in #24
